int8 left
int8 right
---
bool success
